Checkout/Checkout/Source/Validation/Validators/CardValidator.swift

```diff
@@ -175,9 +175,9 @@ public class CardValidator: CardValidating {
   }
 
   public func validate(_ card: Card) -> ValidationResult<ValidationError.Card> {
+    guard let number = card.number else { return .failure(.cardNumber(.invalidCharacters))}
```

Comment:
If we return an error, I think it may be appropriate to declare a correct error for the failure reported, as invalidCharacters is not accurate.

I think we can add another case, `missingNumber` ?

Reply:
Instead of adding new error. Please check why is the card.number even a optional value at first place. Can the number inside the Card struct just be a string.